src/components/DatePickerModal/readme.md

```diff
@@ -1,72 +1,312 @@
 ##### DatePickerModal base:
 
 ```js
-import { Button, DatePickerModal } from 'react-rainbow-components';
+import React from 'react';
+import { Picklist, Option, DatePickerModal } from 'react-rainbow-components';
 
-initialState = { isOpen: false };
+initialState = { isOpen: false, selection: { name: '', label: '' } };
 
-function getDate(value, selectionType) {
+function getDates(value, selectionType) {
     if (selectionType === 'single' && Array.isArray(value)) return value[0];
     return value;
 }
 
-<div
-    className="rainbow-align-content_center rainbow-m-vertical_large rainbow-p-horizontal_small rainbow-m_auto rainbow-flex"
->
-    <div className="rainbow-p-horizontal_medium">
-        <Button
-            label="Click to select a date"
-            onClick={() => setState({ isOpen: true, selectionType: 'single', value: null, title: 'Select a date' })} />
-    </div>
-    <div className="rainbow-p-horizontal_medium">
-        <Button
-            label="Click to select a range of dates"
-            onClick={() => setState({ isOpen: true, selectionType: 'range', value: null, title: 'Select range of dates' })} />
-    </div>
+function setDates(value) {
+    if (value) {
+        if (value.name === 'Custom') {
+            setState({
+                isOpen: true,
+                selectionType: 'range',
+                value: null,
+                title: 'Select a Date',
+            });
+        } else if (value.name === 'Today') {
+            setState({ selection: { name: 'Today', label: 'Today' } });
+        } else if (value.name === 'Yesterday') {
+            setState({ selection: { name: 'Yesterday', label: 'Yesterday' } });
+        } else if (value.name === 'This Week') {
+            setState({ selection: { name: 'This Week', label: 'This Week' } });
+        }
```

Comment:
I think you should only care about the custom since other values come from picklist as they should be stored:
```
if (value.name === 'Custom') {
           return setState({
                isOpen: true,
                value: null,
                title: 'Select a Date',
            });
}
return setState({ selection: value });
```